tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/xml/XmlParserHelper.java

```diff
@@ -120,7 +121,12 @@ public <T> T getJaxbRoot(InputStream stream, Class<T> clazz, Schema schema)
 		ContextProvidingValidationEventHandler handler = new ContextProvidingValidationEventHandler();
 		try {
 			staxEventReader = new JpaNamespaceTransformingEventReader( staxEventReader );
-			JAXBContext jaxbContext = JAXBContext.newInstance( ObjectFactory.class );
+			ClassLoader cl = this.getClass().getClassLoader();
+			Package objectFactoryPackage = ObjectFactory.class.getPackage();
+			if (objectFactoryPackage == null) {
+				throw new AssertionError("Package of ObjectFactory is null and shouldn't be");
+			}
+			JAXBContext jaxbContext = cl == null ? JAXBContext.newInstance( objectFactoryPackage.getName() ) : JAXBContext.newInstance( objectFactoryPackage.getName(), cl );
```

Comment:
```suggestion
			String packageName = NullnessUtil.castNonNull( ObjectFactory.class.getPackage() ).getName();
			JAXBContext jaxbContext = cl == null ? JAXBContext.newInstance( packageName  ) : JAXBContext.newInstance( packageName , cl );
```

Reply:
Done 😄

Reply:
Look, I'm sorry to belabor this, but do we ever expect the annotation processor to be loaded on the bootstrap class loader?

If not, we should throw.

Reply:
We should throw because _I don't know if it's possible or not_.

Reply:
I mean, like, why should the boot classloader being null imply that the correct thing to do is continue with the thread context classloader? Is there any particular justification for that? Perhaps there is, but if there is, I don't know what it is.